### DISCUSSION TITLE
Hard dependency on uvloop

### DISCUSSION THREAD

**Discussion by llob** [canvas-medical/fhirstarter#111] [Ideas]
Hi,

I do my development on Windows, and fhirstarter looks like an awesome project to base our FHIR-compliant integration server on.

fhirstarter however currently has a hard dependency on uvloop, which obviously makes sense for a lot of use cases.
uvloop is not supported on Windows though, so would a pull request which makes uvloop a soft dependency (used if available) have a chance of getting merged?

**Comment by csande** [comment]
I've debated this myself -- I don't think I'd have any issue with that. Go ahead and submit a PR and I'll take a look.

**Reply by csande** [reply]
Also worth mentioning that I've considered removing the `uvloop` dependency entirely -- maybe this library shouldn't be making that call for people. If that is a developer's preference for their project, it's not a big ask to have them add it themselves. Something to consider.

**Reply by llob** [reply]
Ok, I will come up with a PR that removes uvloop entirely then.
Speaking of dependencies, is there a particular reason, that fhirstarter cannot work with python 3.10?

**Reply by csande** [reply]
It would probably work with some minor changes. There's a dependency on `tomllib` which isn't available in 3.10. Aside from that, mostly I went with 3.11 to keep the testing and maintenance burden small. Since it's a framework, rather than a tool that could be integrated into older projects, the main use case would be for new projects that could use 3.11 relatively easily. Also, my expectation would be that it would typically be deployed in containers.

**Comment by llob** [comment]
> Also worth mentioning that I've considered removing the `uvloop` dependency entirely

I have the pretty minimal changes on a local branch. It's pretty straight forward.
 I checked the contribution guidelines, but it doesn't really say anything about how you prefer external contributions? I can't create a feature branch the fhirstarter repo, so do you prefer if I for the repo?

**Reply by csande** [reply]
This is my first attempt at maintaining an open-source library, so I'll probably need to add a bit more to the contributing guidelines.

Others have forked the project and created pull requests on their forks. I recommend that you do that.

**Comment by csande** [comment]
Thank you for the contribution! I will be cutting a release soon that will contain this change.
